### PR TITLE
rsx: Fix unknown Blend equation

### LINKED ISSUE #3357: RSX: Call of Duty 3 - Unknown blend eq 0xffff
Tested by https://github.com/RPCS3/rpcs3/commit/64a6fbe39ff089d0155a36aeb0d3c6360d223ad9

While attemp to ingame it throws fatal error "Unknown blend eq 0xffff"

```
F {rsx::thread} class std::runtime_error thrown: Unknown blend eq 0xffff
(in file Emu\RSX\gcm_enums.cpp:733)
```

**[LOG](https://mega.nz/#!6NEXBTbR!X82qM4kOhbEQvEkamPtxGSEqSI6_V-0EYIxFhacx36Q)**

### LINKED ISSUE #6459: [BLES00148] Call of Duty games - Unknown blend eq 0x0
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3562585/RPCS3.log.gz)
Crash right before getting ingame and compiling couple shaders
If its a duplicate - will close but i saw an issue with blend eq 0x7 only
Also happens in: 
CoD 5 World at War - for ex. when staring new game
CoD Black Ops - for ex. when starting Zombie map "Ascension" (DLC map)
Every game has blend eq 0x0

### PR DESCRIPTION
**Pull request by elad335**
Ignore invalid blend equations in 16-bit wide pieces.

Testcase : https://github.com/elad335/myps3tests/tree/master/rsx_tests/unkown%20Blend%20Equation

Code in practice:
```
cellGcmSetBlendEnable(&Gcm, 1);
cellGcmSetBlendEquation(&Gcm, CELL_GCM_MIN, CELL_GCM_MAX);
cellGcmSetBlendEquation(&Gcm, 0, 0);
```
Results of the crash dump:
```
Equation Color:MIN     Alpha:MAX
```
This means the 0, 0 writes didn't have an effect, essentialy ignoring invalid values.

Fixes #3357
Fixes #6459

### CODE REVIEW

**Comment by kd-11** [comment]
0 is special though.. What about other invalid inputs?

**Comment by elad335** [comment]
Same with (UINT16_MAX, UINT16_MAX)

File: rpcs3/Emu/RSX/rsx_methods.cpp [hunk 1]
```diff
@@ -747,6 +747,37 @@ namespace rsx
 			}
 		}
 
+		void set_blend_eqaution(thread* rsx, u32 reg, u32 arg)
```
Comment (CookiePLMonster) [inline rpcs3/Emu/RSX/rsx_methods.cpp]:
```suggestion
		void set_blend_equation(thread* rsx, u32 reg, u32 arg)
```

File: rpcs3/Emu/RSX/rsx_methods.cpp [hunk 2]
```diff
@@ -2997,6 +3028,7 @@ namespace rsx
 		bind_range<NV4097_SET_VIEWPORT_SCALE, 1, 3, nv4097::set_viewport_dirty_bit>();
 		bind_range<NV4097_SET_VIEWPORT_OFFSET, 1, 3, nv4097::set_viewport_dirty_bit>();
 		bind<NV4097_SET_INDEX_ARRAY_DMA, nv4097::check_index_array_dma>();
+		bind<NV4097_SET_BLEND_EQUATION, nv4097::set_blend_eqaution>();
```
Comment (CookiePLMonster) [inline rpcs3/Emu/RSX/rsx_methods.cpp]:
```suggestion
		bind<NV4097_SET_BLEND_EQUATION, nv4097::set_blend_equation>();
```

**Comment by elad335** [comment]
Fixed a corner case, you are welcone to retest.

**Comment by Xcedf** [comment]
Nice one!
CoD3 300% Scale
0C0xebysygY

**Comment by web1018** [comment]
Capture112


I can confirm that with this PR the game can go ingame, but the game is endlessly compiling shader. I am not sure if thats suppose to happen or not.

**Comment by CookiePLMonster** [comment]
Gran Turismo 5 Prologue now loads more than just two tracks.

**Comment by digitaldude555** [comment]
Screenshot (1372)
Virtua Tennis is now playable.

**Comment by money123451** [comment]
image
COD WAW now goes ingame in the campaign don't use relaxed zcull it causes graphical issues and read color buffers might fix a few other small issues with textures water is still broken and performance is not wonderful with drops to the 20s

**Comment by elad335** [comment]
I tested another combination (colorEquation=CELL_GCM_FUNC_ADD, alphaEquation=UINT16_MAX) and the value was ignored entirely, updating the pr.

**Pull request merged**